internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -594,6 +609,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 		err = errors.Wrap(err, errCompose)
 		r.record.Event(xr, event.Warning(reasonCompose, err))
+		if cm != nil {
+			r.record.Event(cm, event.Warning(reasonCompose, err))
+		}
```

Comment:
I could imagine a special event recorder implementation that did this automatically (e.g. lookup XR's claim and emit an event for that too if there was one). Might be overkill though.